### PR TITLE
[IMP] base_rest: Allow to pass the url parameters as kwargs

### PR DESCRIPTION
**Pull request by albariera**
This PR forward port of #148

### CODE REVIEW

**Review by etobella — APPROVED** [review]

**Comment by lmignon** [comment]
@albariera rest_log overrides the `service._dispatch`. This addon should also be modified into order to be able to merge this PR.
If would also be fine to add a log.debug when args is used to say that `*args`will no more be supported in 15.0 All the rest methods must be adapted to support keyword arguments for path paramters.

**Comment by albariera** [comment]
Done @lmignon! Thank you for your comment :blush:

File: base_rest/components/service.py [hunk 1]
```diff
@@ -149,15 +149,23 @@ def dispatch(self, method_name, *args, params=None):
                        to the method as keyword args.
         :return:
         """
+        if args:
+            _logger.debug(
+                "The use of *args will not be supported in 15.0. "
+                "All rest methods should be adapted to support "
+                "keyword arguments for path paramters."
+            )
```
Comment (lmignon) [inline base_rest/components/service.py]:
```suggestion
                "keyword arguments for path paramters.\n"
                "Service: %s\n"
                "Method: %s\n",
                self._name,
                method_name
            )
```

File: base_rest/restapi.py [hunk 2]
```diff
@@ -38,6 +44,8 @@ def method(
     :param cors: The Access-Control-Allow-Origin cors directive value.
     :param bool csrf: Whether CSRF protection should be enabled for the route.
                       Defaults to ``False``
+    :param bool endpoint_params_to_kwargs: Pass the query parameters as keyword
+                arguments, following the same logic than Werkzeug
```
Comment (lmignon) [inline base_rest/restapi.py:48]:
```suggestion
                arguments, following the same logic than Werkzeug
                Pay attention that this behavior will be the default one into 
                Odoo 15.
          
```

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you for your patience and your hard work. Can you add a test  into base_rest for this new functionality.

File: base_rest/components/service.py (hunk 1 above)
Comment (albariera) [reply base_rest/components/service.py]: Done!

File: base_rest/restapi.py (hunk 2 above)
Comment (albariera) [reply base_rest/restapi.py:48]: Done too :)

**Comment by albariera** [comment]
I have tested using the same logic as the tests already present. Is this what you expected? @lmignon

File: base_rest/tests/test_controller_builder.py [hunk 3]
```diff
@@ -302,6 +310,21 @@ def _get_partner_schema(self):
...
+                "methods": ["POST"],
+                "auth": "user",
+                "cors": None,
+                "csrf": False,
+                "routes": [
+                    "/test_controller/partner/<int:invoice_id>"
+                    "/<string:partner_id>/change_partner"
+                ],
+            },
+        )
```
Comment (lmignon) [inline base_rest/tests/test_controller_builder.py:326]: I think it's possible to test that controller method dispatch correctly the request to the service... 

```python
 from odoo.addons.website.tools import MockRequest
 with MockRequest(self.env):
            res = controller().post_change_partner(invoice_id=...., partner_id=, ...)
            self.assert...

```

**Pull request closed without merging**